assets/src/edit-story/components/checklist/checks/videoOptimization.js

```diff
@@ -0,0 +1,151 @@
+    element.resource?.isOptimized
+  ) {
+    return false;
+  }
+
+  const videoArea =
+    (element.resource?.height ?? 0) * (element.resource?.width ?? 0);
+  const isLargeVideo =
+    videoArea >= VIDEO_SIZE_THRESHOLD.WIDTH * VIDEO_SIZE_THRESHOLD.HEIGHT;
+  return isLargeVideo;
+}
+
+export const BulkVideoOptimization = () => {
+  const story = useStory(({ state }) => state);
+  const unoptimizedVideos = filterStoryElements(story, videoElementOptimized);
+
+  const setHighlights = useHighlights(({ setHighlights }) => setHighlights);
+
+  const { optimizeVideo } = useLocalMedia((state) => ({
+    optimizeVideo: state.actions.optimizeVideo,
+  }));
+
+  const [isOptimizing, setIsOptimizing] = useState(false);
+
+  const sequencedVideoOptimization = useCallback(() => {
+    return new Promise((res) => {
+      unoptimizedVideos
+        .filter(({ resource }) => !resource?.isTranscoding)
+        .map(({ resource }) => async () => {
+          await optimizeVideo({ resource });
+        })
+        .reduce((p, fn) => p.then(fn), Promise.resolve())
+        .then(() => {
+          res();
+        });
+    });
+  }, [unoptimizedVideos, optimizeVideo]);
+
+  const handleUpdateVideos = useCallback(async () => {
+    setIsOptimizing(true);
+    await sequencedVideoOptimization();
+    setIsOptimizing(false);
+  }, [sequencedVideoOptimization]);
+
+  const { footer, title } = PRIORITY_COPY.videoNotOptimized;
+
+  const isTranscoding = useMemo(
+    () =>
+      isOptimizing ||
+      unoptimizedVideos.some((video) => video.resource?.isTranscoding),
+    [isOptimizing, unoptimizedVideos]
+  );
+  return (
+    unoptimizedVideos.length > 0 && (
+      <ChecklistCard
+        cta={
+          <OptimizeButton
+            type={BUTTON_TYPES.TERTIARY}
+            size={BUTTON_SIZES.SMALL}
+            onClick={handleUpdateVideos}
+            disabled={isTranscoding}
+          >
+            {isTranscoding
+              ? __('Optimizing videos', 'web-stories')
+              : __('Optimize all videos', 'web-stories')}
+          </OptimizeButton>
+        }
+        title={title}
+        cardType={
+          unoptimizedVideos.length > 1
+            ? CARD_TYPE.MULTIPLE_ISSUE
+            : CARD_TYPE.SINGLE_ISSUE
+        }
+        footer={<DefaultFooterText>{footer}</DefaultFooterText>}
+        thumbnailCount={unoptimizedVideos.length}
+        thumbnail={
+          <>
+            {unoptimizedVideos.map((element) => (
+              <Thumbnail
+                key={element.id}
+                onClick={() => {
+                  setHighlights({
+                    elementId: element.id,
+                  });
+                }}
+                isLoading={element.resource?.isTranscoding}
+                type={THUMBNAIL_TYPES.VIDEO}
+                displayBackground={<LayerThumbnail page={element} />}
+                aria-label={__('Go to offending video', 'web-stories')}
+              />
+            ))}
+          </>
```

Comment:
I know you've specified "bulk" for this, so feel free to have this be separate, but the video optimization chard needs this 💅  extra flare for when you hover or focus the card, I already outlined the necessary code for it in storybook, just adding: 
```
<Thumbnail yadda={yadda}>
        <Tooltip title={__('Optimize", "web-stories")}>
          <StyledVideoOptimizationIcon />
        </Tooltip>
      )}
    </Thumbnail>
    ```

<img width="150" alt="Screen Shot 2021-06-25 at 9 08 43 AM" src="https://user-images.githubusercontent.com/10720454/123454036-f5c2ea00-d594-11eb-859a-07f2a3680cbf.png">